### ISSUE TITLE
[css-sizing] Intrinsic size of replaced elements incorrect

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#794]
Currently Sizing 3 defines intrinsic sizes of replaced elements explicitly.
This definition isn't correct, and it should be deleted in favor of falling through to the CSS2.1 reference used for non-replaced elements.

**Comment by fantasai** [comment]
Agenda+ to point out that the implication here is that, when an aspect ratio is present, `min-content` and `max-content` are dependent on sizing constraints in the opposite axis. (This might be surprising/disturbing to implementers but is probably as expected for authors.)

**Comment by dbaron** [comment]
I really don't want <code>min-content</code> and <code>max-content</code> to be anything other than *intrinsic* sizes.  And, in particular, I want them never to depend on the element's own specified sizes, precisely so that they can be used as keywords for the element's specified sizes.

Doing so could cause problems down the road if we ever add the ability for calc() expressions to reference concepts like "this element's <code>min-content</code> size in the other dimension" or "the grandparent element's size".  (The latter is harder than the former, but we might be able to come up with a set of ways where it does make sense or a way of resolving the constraints, and having things like "intrinsic" sizes that are almost but not quite always intrinsic would make that harder.)

**Comment by fantasai** [comment]
The problem with that is then we don't have a keyword for "shrinkwrap this thing as if it was a float", which is what you generally want in e.g. grid track sizing.

**Comment by dbaron** [comment]
I just discussed this a bit with @fantasai.

To clarify what I want instead:  I'd like the `width` and the `height` to both be considered when computing an element's min-content contribution to its parent.  Right now (assuming we're talking about English layout, i.e., `min-content` as a `width`), it's clear that for `width`, an element's `min-content` width doesn't consider its `width` property (this allows things like `width: min-content`).  What we're disagreeing about is whether this is the same for how `height` is considered, on elements that have an intrinsic ratio.  I'd like the `height` to be considered at the same stage as `width`, whereas @fantasai wants it to be considered earlier, so that it's incorporated into what the `min-content` value means.

I don't think the grid track sizing thing is an issue, since that should be depending on the min-content contribution, which is affected by the aspect ratio with either proposal.

@fantasai's approach has the advantage (for the spec authors, and *maybe* implementors) that defining what `width: auto` means is much simpler.  With my approach, the meaning of `width: auto` is more complicated since it has special rules for replaced elements.

There isn't a distinct author feature in @fantasai's proposal, I don't think; `auto` will still work the way she wants `min-content` and `max-content` to behave.  So with what I'm proposing there are more behaviors exposed to authors, although they seem unlikely to be useful (although I believe they are underlying primitives).

I still think my proposal is more future-proof for features like the ones I described above in https://github.com/w3c/csswg-drafts/issues/794#issuecomment-268181106
